clang/lib/APINotes/APINotesYAMLCompiler.cpp

```diff
@@ -730,7 +731,9 @@ class YAMLConverter {
     }
   }
 
-  void convertParams(const ParamsSeq &Params, FunctionInfo &OutInfo) {
+  std::optional<ParamInfo> convertParams(const ParamsSeq &Params,
+                                         FunctionInfo &OutInfo) {
+    std::optional<ParamInfo> thisOrSelf;
```

Comment:
What do you think of `instance` as the name instead?

Reply:
In Objective-C static methods, `self` is... the instance of the class object. In this case, `instance` isn't exactly wrong---the `self` is an instance of something---but it's a little misleading because it's not an instance of the enclosing type.

Reply:
Leaving it as is, as per Doug's point.